src/transport/mqtt/connection.rs

```diff
@@ -488,26 +531,58 @@ impl Handshake {
         session_data: SessionData,
     ) -> Next
     where
-        S: PropertyStore,
+        S: PropertyStore + StoreCapabilities,
     {
         let client = conn.client.clone();
+        let client_id: ClientId = client_id.into();
+        let store = store.clone();
 
-        let client_id_cl: ClientId = client_id.into();
+        let handle = if self.session_present && conn.session_sync {
+            info!(
+                "Already synchronized skipping handshake (introspection synchronized: {})",
+                conn.session_sync
+            );
 
-        let store = store.clone();
-        let handle: JoinHandle<Result<(), InitError>> = tokio::spawn(async move {
-            let client_id = client_id_cl.as_ref();
+            Self::prop_handshake(client, client_id, store, session_data, self.session_present)
+        } else {
+            info!(
+                "The device is not synchronized (session_present: {}, matching introspection: {}), redoing handshake",
```

Comment:
also here, something like:

```suggestion
                "Perform again handshake to synchronize the device",
```

the actual log could be good in DEBUG mode, but imo it's too long for INFO level